posthog/models/notification_setting.py

```diff
@@ -0,0 +1,46 @@
+from enum import Enum
+from typing import Optional
+
+from posthog.models import User
+
+
+class NotificationSetting(Enum):
+    WEEKLY_PROJECT_DIGEST = "weekly_project_digest"
+    PLUGIN_DISABLED = "plugin_disabled"
+    # Add more notification types as needed
+
+
+def should_send_notification(
+    user: User,
+    notification_type: NotificationSetting,
```

Comment:
This is a `str` since you're passing in `NotificationSetting.WEEKLY_PROJECT_DIGEST.value` so you can either update this or you can update the value being passed in to be `NotificationSetting.WEEKLY_PROJECT_DIGEST` and the if statement below to do the same.